models/playbooks.go

```diff
@@ -0,0 +1,44 @@
+package models
+
+import (
+	"time"
+
+	"github.com/flanksource/duty/types"
+	"github.com/google/uuid"
+)
+
+type PlaybookRunStatus string
+
+const (
+	PlaybookRunStatusScheduled PlaybookRunStatus = "scheduled"
+	PlaybookRunStatusRunning   PlaybookRunStatus = "running"
+	PlaybookRunStatusCancelled PlaybookRunStatus = "cancelled"
+	PlaybookRunStatusFailed    PlaybookRunStatus = "failed"
+	PlaybookRunStatusCompleted PlaybookRunStatus = "completed"
+)
+
+type Playbook struct {
+	ID        uuid.UUID  `gorm:"default:generate_ulid()"`
+	Name      string     `json:"name"`
+	Spec      types.JSON `json:"spec"`
+	CreatedBy *uuid.UUID `json:"created_by,omitempty"`
+	CreatedAt time.Time  `json:"created_at,omitempty" time_format:"postgres_timestamp"`
+	UpdatedAt time.Time  `json:"updated_at,omitempty" time_format:"postgres_timestamp"`
+	DeletedAt *time.Time `json:"deleted_at,omitempty" time_format:"postgres_timestamp"`
+}
+
+type PlaybookRun struct {
+	ID          uuid.UUID         `gorm:"default:generate_ulid()"`
+	PlaybookID  uuid.UUID         `json:"playbook_id"`
+	CreatedAt   time.Time         `json:"created_at,omitempty" time_format:"postgres_timestamp" gorm:"default:NOW()"`
+	StartDate   time.Time         `json:"start_date,omitempty" time_format:"postgres_timestamp" gorm:"default:NOW()"`
+	EndDate     *time.Time        `json:"end_date,omitempty" time_format:"postgres_timestamp"`
+	Duration    time.Duration     `json:"duration" gorm:"default:null"`
+	Result      types.JSON        `json:"result,omitempty"`
+	CreatedBy   *uuid.UUID        `json:"created_by,omitempty"`
+	ComponentID *uuid.UUID        `json:"component_id,omitempty"`
+	ConfigID    *uuid.UUID        `json:"config_id,omitempty"`
```

Comment:
Can a run have multiple of these ids ?

Reply:
A run can have only one of either config or the component id.